app/src/main/java/org/wikipedia/settings/AppIconDialog.kt

```diff
@@ -0,0 +1,138 @@
+package org.wikipedia.settings
+
+import android.content.res.ColorStateList
+import android.os.Build
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.content.ContextCompat
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.flexbox.AlignItems
+import com.google.android.flexbox.FlexDirection
+import com.google.android.flexbox.FlexboxLayoutManager
+import com.google.android.flexbox.JustifyContent
+import org.wikipedia.LauncherController
+import org.wikipedia.LauncherIcon
+import org.wikipedia.R
+import org.wikipedia.appshortcuts.AppShortcuts
+import org.wikipedia.databinding.DialogAppIconBinding
+import org.wikipedia.databinding.ItemAppIconBinding
+import org.wikipedia.page.ExtendedBottomSheetDialogFragment
+import org.wikipedia.util.FeedbackUtil
+
+class AppIconDialog : ExtendedBottomSheetDialogFragment() {
+    private var _binding: DialogAppIconBinding? = null
+    private val binding get() = _binding!!
+
+    private val appIconAdapter: AppIconAdapter by lazy {
+        AppIconAdapter().apply {
+            onItemClickListener { selectedIcon ->
+                Prefs.currentSelectedAppIcon = selectedIcon.key
+                LauncherController.setIcon(selectedIcon)
+                AppShortcuts.setShortcuts(requireContext())
+                updateIcons(selectedIcon)
+                FeedbackUtil.makeSnackbar(
+                    binding.root,
+                    "App icon changed successfully.").show()
+            }
```

Comment:
Can we dismiss the bottomsheet after selecting the icon? the snackbar message will overlap the bottomsheet now.

You can use `FeedbackUtil.makeSnackbar(requireActivity(), "App icon changed successfully.").show()` to prevent a possible crash.

Reply:
done.